internal/util/ctxutil/ctxutil_test.go

```diff
@@ -42,4 +42,38 @@ func TestDurationWithJitter(t *testing.T) {
 		sleep := DurationWithJitter(time.Second, 0)
 		assert.LessOrEqual(t, sleep, time.Second)
 	})
+
+	t.Run("multiple tasks retry multiple times", func(t *testing.T) {
+		// This test outputs a file for duration it took all nTasks to retry nRetries.
+		// In reality not all tasks will retry, but this is good enough for visualising it.
+
+		nTasks := 100
+		nRetries := 5
+
+		durations := make([][]time.Duration, nTasks) // task -> retry count -> duration
+
+		for i := 0; i < nTasks; i++ {
+			durations[i] = make([]time.Duration, nRetries)
+			for j := 0; j < nRetries; j++ {
+				durations[i][j] = DurationWithJitter(time.Second, int64(j+1))
+			}
+		}
+
+		dir := filepath.Join("result")
+		err := os.MkdirAll(dir, 0o666)
+		require.NoError(t, err)
+
+		filename := filepath.Join(dir, "multiple-retry-jitter.txt")
+		f, err := os.Create(filename)
+		require.NoError(t, err)
+
+		defer f.Close()
+
+		for _, task := range durations {
+			for j, duration := range task {
+				// each line has retry count (j+1) and duration waited in milliseconds.
+				fmt.Fprintln(f, j+1, duration.Milliseconds())
+			}
+		}
+	})
```

Comment:
Simple plot of output. This uses 100 tasks all retrying 5 times, yes unrealistic but it's good enough to visualised it.
The first retry is capped ~200ms (exponential part). After 4th attempt, the exponential part has reached to `cap` so it's no more exponential backoff. The scatter shows that jitter is working well.

![image](https://user-images.githubusercontent.com/24355139/232435450-0139a525-9336-4338-81c2-fa9ecfaa2557.png)